### ISSUE TITLE
MqttClient.Connect() hangs when there is no connection

### ISSUE THREAD

**Issue by imetallica** [mohaqeq/M2MqttDotnetCore#2]
Hey, when I try to call `connect` when there is no connection, the function hangs, even when the MQTT goes back online.

Example (with F#)
```
let conn = MqttClient(IPAddress.Parse("127.0.0.1"))
let result = conn.Connect("hello-world") // <-- Hangs here.
```
 
I'm using dotnetcore 2.0

**Comment by mohaqeq** [comment]
Set the timeout and try again.

**Comment by imetallica** [comment]
Can you give a small example? I couldn't find any documentation about it.

Enviado do meu iPhone

**Comment by mohaqeq** [comment]
There is class named MqttSettings.
You can set the default connection time out in the singleton instance "MqttSettings.Instance" before creating client.

**Comment by imetallica** [comment]
Yeah, that would work if the modification access wasn't internal.

```
...
public int TimeoutOnConnection { get; internal set; }
...
```

Can you change the access modification and push a new version to nuget?

**Comment by mohaqeq** [comment]
I know it is internal. Please clone this code and modify it to check if it solve your problem.
